### PR TITLE
Improve browser manipulations mechanism. Update testcafe-browser-tools (fixes #1732, fixes #1551)

### LINKED ISSUE #1551: Screenshoting fails on Windows because screenshot directory was not created
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
Screenshot native module fails to save screenshot at the specified path, because C# API reports that some subdirectories of the path were not created, despite they seem to be created from Node.js side. Adding a delay btw. directory creation from Node.js and taking screenshot helps.

### What is the expected behavior?
Should take the screenshot.

### Specify your

* operating system: Windows 8.1
* testcafe version: 0.16.0

### LINKED ISSUE #1732: The test can’t run remote device
### Are you requesting a feature or reporting a bug?

bug

### What is the current behavior?

Well, two days ago, I can run the test on the remote device, but it can't work now.And I didn't edit the the test scripts.

So I try to use the get started demo code you provided on the website. And type

```js
testcafe remote demo.js -e
```

But it can't run on the remote device, I get the req and res through charles,I find that it doesn't load the test page and stay in the init page:

wechatimg13

But I run on the chrome:

```js
test chrome demo.js -e
```

It works well, and load the demo test page.

### What is the expected behavior?

It can run on remote device correctly.

### How would you reproduce the current behavior (if this is a bug)?

I use the test demo code:

```javascript
import { Selector } from 'testcafe';

fixture `Getting Started`
    .page `http://devexpress.github.io/testcafe/example`;

test('My first test', async t => {
    await t
        .typeText('#developer-name', 'John Smith')
        .click('#submit-button');
});
```

and run it :

```javascript
test remote demo.js -e
```

No matter I open the Connect URL on the remote device or desktop chrome, it can't load the test page and stay in the initial page.But If run it :

```js
test chrome demo.js -e
```

It works well

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```js

```

### Specify your

* operating system: Mac OS X 10.11
* testcafe version: 0.17.0
* node.js version: 8.0.0

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
Fixes: 
 * #1732 
 * #1551 
 * #1625 ?

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
{ Error: Command failed: C:\projects\testcafe\node_modules\testcafe-browser-tools\bin\win\generate-thumbnail.exe __test-screenshots___\2017-08-11_05-46-54\test-1\run-2\Chrome_60.0.3112_Windows_8.1_0.0.0\errors\1.png __test-screenshots___\2017-08-11_05-46-54\test-1\run-2\Chrome_60.0.3112_Windows_8.1_0.0.0\errors\thumbnails\1.png 240 130
Error 79: failed to open file for writing
    at ChildProcess.exithandler (child_process.js:270:12)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:921:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
  killed: false,
  code: 1,
  signal: null,
  cmd: 'C:\\projects\\testcafe\\node_modules\\testcafe-browser-tools\\bin\\win\\generate-thumbnail.exe __test-screenshots___\\2017-08-11_05-46-54\\test-1\\run-2\\Chrome_60.0.3112_Windows_8.1_0.0.0\\errors\\1.png __test-screenshots___\\2017-08-11_05-46-54\\test-1\\run-2\\Chrome_60.0.3112_Windows_8.1_0.0.0\\errors\\thumbnails\\1.png 240 130' }

**Comment by AndreyBelym** [comment]
FPR

File: test/functional/fixtures/api/es-next/resize-window/test.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var config                     = require('../../../../config.js');
 var errorInEachBrowserContains = require('../../../../assertion-helper.js').errorInEachBrowserContains;
 
 
-describe('[API] Resize window actions', function () {
+describe.only('[API] Resize window actions', function () {
```
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/api/es-next/resize-window/test.js]: missed `only`

File: src/client/driver/driver.js [hunk 2]
```diff
@@ -379,7 +379,7 @@ export default class Driver {
     _onPrepareBrowserManipulationCommand () {
         this.contextStorage.setItem(this.COMMAND_EXECUTING_FLAG, true);
 
-        prepareBrowserManipulation(this.browserId)
+        prepareBrowserManipulation()
```
Comment (AlexanderMoskovkin) [inline src/client/driver/driver.js]: it seems we can get rid of `this.browserId` now: https://github.com/DevExpress/testcafe/blob/master/src/client/driver/driver.js#L83

**Pull request merged**